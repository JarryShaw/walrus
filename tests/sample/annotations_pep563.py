from __future__ import annotations

import typing


def foo() -> (a := int):
    return 1


print(foo())
print(type(foo.__annotations__['return']))
print(globals().get('a'))
print(typing.get_type_hints(foo))
print(a)
