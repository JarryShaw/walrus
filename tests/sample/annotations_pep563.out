1
<class 'str'>
None
{'return': <class 'int'>}
<class 'int'>
